pkg/controllers/clusterinventory/clusterprofile/controller_test.go

```diff
@@ -0,0 +1,157 @@
+							},
+							LastReceivedHeartbeat: metav1.Time{Time: time.Now().Add(-10 * clusterUnhealthyThreshold)},
+						},
+					},
+				},
+			},
+			clusterProfile:          &clusterinventory.ClusterProfile{},
+			expectedConditionStatus: metav1.ConditionUnknown,
+			expectedConditionReason: clusterHeartbeatLostReason,
+		},
+		{
+			name: "Member agent health check result is out of date or unknown",
+			memberCluster: &clusterv1beta1.MemberCluster{
+				Status: clusterv1beta1.MemberClusterStatus{
+					AgentStatus: []clusterv1beta1.AgentStatus{
+						{
+							Type: clusterv1beta1.MemberAgent,
+							Conditions: []metav1.Condition{
+								{
+									Type:   string(clusterv1beta1.AgentHealthy),
+									Status: metav1.ConditionUnknown,
+								},
+							},
+							LastReceivedHeartbeat: metav1.Time{Time: time.Now().Add(-1 * time.Second)},
+						},
+					},
+				},
+			},
+			clusterProfile:          &clusterinventory.ClusterProfile{},
+			expectedConditionStatus: metav1.ConditionUnknown,
+			expectedConditionReason: clusterHealthUnknownReason,
+		},
+		{
+			name: "Member agent reports that the API server is unhealthy",
+			memberCluster: &clusterv1beta1.MemberCluster{
+				Status: clusterv1beta1.MemberClusterStatus{
+					AgentStatus: []clusterv1beta1.AgentStatus{
+						{
+							Type: clusterv1beta1.MemberAgent,
+							Conditions: []metav1.Condition{
+								{
+									Type:   string(clusterv1beta1.AgentHealthy),
+									Status: metav1.ConditionFalse,
+								},
+							},
+							LastReceivedHeartbeat: metav1.Time{Time: time.Now().Add(-1 * time.Second)},
+						},
+					},
+				},
+			},
+			clusterProfile:          &clusterinventory.ClusterProfile{},
+			expectedConditionStatus: metav1.ConditionFalse,
+			expectedConditionReason: clusterUnHealthyReason,
+		},
+		{
+			name: "Member agent reports that the API server is healthy",
+			memberCluster: &clusterv1beta1.MemberCluster{
+				Status: clusterv1beta1.MemberClusterStatus{
+					AgentStatus: []clusterv1beta1.AgentStatus{
+						{
+							Type:                  clusterv1beta1.MemberAgent,
+							LastReceivedHeartbeat: metav1.Time{Time: time.Now().Add(-1 * time.Second)},
+							Conditions: []metav1.Condition{
+								{
+									Type:   string(clusterv1beta1.AgentHealthy),
+									Status: metav1.ConditionTrue,
+								},
+							},
+						},
+					},
+				},
+			},
+			clusterProfile:          &clusterinventory.ClusterProfile{},
+			expectedConditionStatus: metav1.ConditionTrue,
+			expectedConditionReason: clusterHealthyReason,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reconciler.syncClusterProfileCondition(tt.memberCluster, tt.clusterProfile)
+			condition := meta.FindStatusCondition(tt.clusterProfile.Status.Conditions, clusterinventory.ClusterConditionControlPlaneHealthy)
+			if condition == nil {
+				t.Fatalf("expected condition to be set, but it was not")
+			}
+			if condition.Status != tt.expectedConditionStatus {
```

Comment:
Hi Ryan! It might be easier to use the cmp package?

Reply:
will open another PR